app/queries/student_details_resolver.rb

```diff
@@ -55,15 +54,15 @@ def course
   end
 
   def authorized?
-    return false if current_user.faculty.blank?
+    return false if current_user.blank?
 
     return false if student.blank?
 
-    current_user.faculty.reviewable_courses.where(id: student.course).exists?
+    current_user.id == student.user.id || current_user.faculty.reviewable_courses.where(id: student.course).exists?
```

Comment:
1. `current_user.id == student.user_id`. Don't load the related record unnecessarily - there's a preload for this in the `student` method.
2. Ditch `reviewable_courses`.